Moving default-policy.json and default.yaml to containers/image

rahilarious opened the pull request:
All major distros (Fedora, debian, archlinux, gentoo, alpine, opensuse) are placing these 2 files in containers-common package. Why not fix it upstream?

1st step towards addressing https://github.com/containers/skopeo/issues/2170

mtrmac reviewed (commented):
I have no opinion on the location of these files, I’ll fully defer to our packaging experts (and this _must_ be coordinated with them).

@lsm5 @jnovy PTAL.

---

I do have one opinion: Skopeo’s tests must continue to work, and this causes a failure.

mtrmac commented on .cirrus.yml:
```diff
@@ -147,6 +147,7 @@ ostree-rs-ext_task:
         - dnf builddep -y skopeo
         - make
         - make install
+        - echo '{"default":[{"type":"insecureAcceptAnything"}],"transports":{"docker-daemon":{"":[{"type":"insecureAcceptAnything"}]}}}' > /etc/containers/policy.json
```
- I’m afraid this is failing
- I’m tempted to say that this should do the same thing we are asking other users to do. I.e. if users are expected to install the files from the c/image repo, this should also go through the troubler of installing the files from the c/image repo.

rahilarious replied:
>     * I’m tempted to say that this should do the same thing we are asking other users to do. I.e. if users are expected to install the files from the c/image repo, this should also go through the troubler of installing the files from the c/image repo.

I agree. I just don't know how to handle cirrus stuff. It's up to you to properly fix it.

mtrmac replied:
It is… just a shell script?

---

… Uh. The one thing we _might_ need to provide some guidance on is to figure out which c/image version to use. ~Luckily the data is available in `go.mod`, but turning that into a `git` command is a bit cumbersome.

@lsm5 @jnovy @rhatdan What do we do for Podman users who build from source, and need `containers.conf` from c/common? It seems to be either “nothing and let them figure it out” or “install a previous packaged of `containers-common` and hope the files match”, is there something I’m missing?

rhatdan replied:
Nothing but let them figure it out.  Luckily Podman runs fine with no containers.conf and we encourage distributions to comment out containers.conf to be used information purposes only, IE Document the default in the system file, and allow admin and users to customize individual fields.

mtrmac replied:
I’m afraid `policy.json` is mandatory. Oh well… I’m always happy to recommend using packaged versions :)

mtrmac commented:
For the record, the files, and the Make rules to install them, were moved to https://github.com/containers/image/pull/2215 .

rahilarious commented:
> @mtrmac I guess we can revise this PR to handle any outstanding cleanup issues if @rahilarious is willing to continue this.

I'm unfamiliar with cirrus stuff, any help would be appreciated

rahilarious committed:
Migrating default-policy.json & default.yaml to container-libs/image

a step towards addressing
https://github.com/podman-container-tools/skopeo/issues/2170

all these files and a dir is already being installed by
containers-common package which skopeo depends on. No need to install by skopeo

Signed-off-by: Rahil Bhimjiani <me@rahil.rocks>

rahilarious committed:
Makefile: no build while installing,build completions in all: target

generate completions along with docs when runnning `make all` & don't
build binary while installing anything. Installation phase should only
install and compile phase should only compile

this makes `make` && `make install` sufficient

change rpm spec file accordingly too

Signed-off-by: Rahil Bhimjiani <me@rahil.rocks>

mtrmac commented on Makefile:
```diff
@@ -159,13 +152,8 @@ clean:
-	install -m 644 default.yaml ${DESTDIR}${REGISTRIESDDIR}/default.yaml
 
-install-binary: bin/skopeo
+install-binary:
```
The closest thing there is to a standard for this, https://www.gnu.org/prep/standards/html_node/Standard-Targets.html , specifies that `install` should build if necessary. Admittedly our use of phony targets for the things to be installed is very suboptimal.

Either way I don’t know why the changes to `install-binary` or `install-completions` or `skopeo.spec` should be a part of this PR at all.

rahilarious replied:
> Either way I don’t know why the changes to `install-binary` or `install-completions` or `skopeo.spec` should be a part of this PR at all

You're right, at least by PR title they shouldn't be there. Should I create new PR for those or change title mentioning both changes?

mtrmac commented on .cirrus.yml:
```diff
@@ -160,6 +160,7 @@ ostree-rs-ext_task:
         - dnf builddep -y skopeo
         - make
         - make install
+        - echo '{"default":[{"type":"insecureAcceptAnything"}],"transports":{"docker-daemon":{"":[{"type":"insecureAcceptAnything"}]}}}' > /etc/containers/policy.json
```
> as per https://github.com/podman-container-tools/container-libs/blob/main/common/rpm/containers-common.spec , these files are being installed from `container-libs/image` repo so this PR should be merged now.

Well then the CI stanza that adds a hard-coded policy shouldn’t be needed, and shouldn’t be part of the PR?!

(In the meantime, we have migrated the CI from `.cirrus.yml` to GitHub actions; that makes the `.cirrus.yml` change ineffective, but either way the PR should not be adding that line.)

rahilarious replied:
done. Removed that line

mtrmac reviewed (changes requested):
https://github.com/podman-container-tools/skopeo/blob/main/install.md documents how users can use `make install`, and that would now be _insufficient_ to obtain a working installation; so that must be extended to also add the other required steps … and that gets us back to https://github.com/podman-container-tools/skopeo/pull/2173#discussion_r1443011122 .

rahilarious committed:
Merge branch 'main' into default-yaml-policy-json-to-common

mtrmac reviewed (changes requested):
https://github.com/podman-container-tools/skopeo/pull/2173#pullrequestreview-4765932069 still applies AFAICT.
